cpp/src/arrow/flight/sql/odbc/tests/connection_info_test.cc

```diff
@@ -239,58 +240,59 @@ TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_DRIVER_HLIB) {
 }
 
 // These information types are implemented by the Driver Manager alone.
-TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_DRIVER_HSTMT) {
-  // TODO This is failing due to no statement being created
-  // This should run after SQLGetStmtAttr is implemented
-  GTEST_SKIP();
+TYPED_TEST(FlightSQLODBCTestBase, TestSQLGetInfoDriverHstmt) {
   this->connect();
 
-  validate(this->conn, SQL_DRIVER_HSTMT, static_cast<SQLULEN>(0));
+  // Value returned from driver manager is the stmt address
+  SQLHSTMT local_stmt = this->stmt;
+  SQLRETURN ret = SQLGetInfo(this->conn, SQL_DRIVER_HSTMT, &local_stmt, 0, 0);
+  EXPECT_EQ(ret, SQL_SUCCESS);
+  EXPECT_GT(local_stmt, static_cast<SQLHSTMT>(0));
 
   this->disconnect();
```

Comment:
Added a test case fix here as well. I needed to call `SQLGetInfo` directly since `SQL_DRIVER_HSTMT` requires the input `this->stmt` according to the spec: https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlgetinfo-function?view=sql-server-ver17 cc @rscales